src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,488 @@
+<?php
+/**
+ * Autoloader for WordPress classes.
+ *
+ * Include this file if you'd like to avoid having to create your own autoloader.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Autoloader.
+ */
+final class WP_Autoload {
+
+	/**
+	 * An array of classes and their respective paths.
+	 *
+	 * Note: The classnames should be lowercased.
+	 *
+	 * @access private
+	 *
+	 * @var array
+	 */
+	const CLASSES_PATHS = array(
+		/* Classes in the wp-includes/ folder. */
+		'passwordhash'                                => 'wp-includes/class-phpass.php',
```

Comment:
```suggestion
		'passwordhash'                                => WPINC . '/class-phpass.php',
```
Should we do this for all these?

Reply:
I'd rather keep it simple in that array... Not all classes are in `wp-includes` (there are many in `wp-admin` as well), so for consistency and visual parity I'd say we can keep them hardcoded.
The `WPINC` constant is not something that users can override anyway, it's only part of Core for convenience 👍

Reply:
I wonder if there is a performance benefit of using a const here? See https://stackoverflow.com/questions/51959743/php-performance-of-using-constants-instead-of-strings

Reply:
If the whole string was a const, then it would make sense... But in this case, we have both strings and the WPINC const. If we switch to using the constant, then we won't be improving anything since we'll have a mix of constants, strings, and the performance hit from concatenating those to get the final string 😬